src/opendr/engine/data.py

```diff
@@ -211,15 +211,45 @@ def __str__(self):
 class Image(Data):
     """
     A class used for representing image data.
+    OpenDR uses CHW/RGB conventions
     This class provides abstract methods for:
     - returning a NumPy compatible representation of data (numpy())
+    - loading an input directly into OpenDR compliant format (open())
+    - getting an image into OpenCV-compliant format (opencv()) for visualization purposes
     """
 
-    def __init__(self, data=None, dtype=np.uint8):
+    def __init__(self, data=None, dtype=np.uint8, guess_format=True):
+        """
+        Image constructor
+        :param data: Data to be held by the image object
+        :type data: numpy.ndarray
+        :param dtype: type of the image data provided
+        :type data: numpy.dtype
+        :param guess_format: try to automatically guess the type of input data and convert it to OpenDR format
+        :type guess_format: bool
+        """
         super().__init__(data)
 
         self.dtype = dtype
         if data is not None:
+            # Check if the image is in the correct format
+            try:
+                data = np.asarray(data)
+            except Exception:
+                raise ValueError("Image data not understood (cannot be cast to NumPy array).")
+
+            if data.ndim != 3:
+                raise ValueError("3D dimensional images are expected")
+            if guess_format:
+                # If channels are found last and image is a color one, assume OpenCV format
+                if data.shape[2] == 3:
+                    data = cv2.cvtColor(data, cv2.COLOR_BGR2RGB)
+                    data = np.transpose(data, (2, 0, 1))
+                # If channels are found last and image is not a color one, just perform transpose
+                elif data.shape[2] < min(data.shape[0], data.shape[1]):
+                    data = np.transpose(data, (2, 0, 1))
+            else:
+                raise ValueError("Image is of type None")
```

Comment:
@passalis This line has to be moved one level up. I realized this when using the method with `guess_format=False`.

Reply:
Thanks for catching another one bug! If pushed a fix for this.